packages/client/src/runtime/utils/waitForBatch.ts

```diff
@@ -0,0 +1,36 @@
+import { hasBatchIndex } from '@prisma/engine-core'
+
+/**
+ * Waits for result of batch $transaction and picks the best possible error to report if any
+ * of the request fails. Best error is determined as follows:
+ *
+ * - if engine have reported and index of failed request in the batch, the best error is the one
+ * who's index matches that
+ * - otherwise, first error is used (like Promise.all)
```

Comment:
Do you know if it would be possible to widen this feature to `Promise.all` (if not already)? If I understood correctly, it uses the same batching API, so do you think we could use the DataLoader/Engine code to implement it for both?